### ISSUE TITLE
Commits cannot be shown with complex trees

### ISSUE THREAD

**Issue by delphinus** [lusingander/serie#46]
When opened repos that have complex and long history, it shows wide blank space and appears no commits. This image below is for [neovim/neovim](https://github.com/neovim/neovim).

Can serie limit the width for trees and/or scroll horizontally not to hide commits?

<img width="512" alt="スクリーンショット 2024-09-06 10 53 38" src="https://github.com/user-attachments/assets/34e4f7d4-306e-46c0-8f87-2791e64588b1">

**Comment by lusingander** [comment]
Yes, this is something I'd like to improve, but I didn't have any good ideas.

I thought [horizontal scrolling would be difficult](https://github.com/lusingander/serie/issues/4#issuecomment-2254957135), but I've [changed the internal implementation](https://github.com/lusingander/serie/pull/38) so it might be possible to implement something practical.

**Comment by alerque** [comment]
Perhaps the the column where the commit info is drawn could not be static (the width of the widest part of the graph of the entire history) but instead dynamic, and relative to the widest part of the graph that is actually in view.

**Comment by daeho-ro** [comment]
`serie` requests the horizontal size 537 in my one of repo...

**Comment by FloppyDisco** [comment]
just downloaded this. super cool tool. 
my immediate thought was like @alerque:

the space reserved for the graph should only take up what is needed based on the width of the graph currently visible, if you scroll down and a new graph like appears the commit section should move over to make room. and if one disappears it can move back left to show more of the commits.

i'd love to help out on this project! very cool tool.
